pkg/chains/storage/oci/legacy.go

```diff
@@ -119,12 +119,22 @@ func (b *Backend) uploadSignature(ctx context.Context, format simple.SimpleConta
 	imageName := format.ImageName()
 	logger.Infof("Uploading %s signature", imageName)
 
-	ref, err := newDigest(b.cfg, imageName)
+	var opts []name.Option
+	if b.cfg.Storage.OCI.Insecure {
+		opts = append(opts, name.Insecure)
+	}
```

Comment:
@lcarva  Not sure if this would be the right method to make sure the insecure option is applied even if cfg.Storage.OCI.Repository == "".

Reply:
In chat, I had suggested something like this:

```go
func newRepo(cfg config.Config, imageName name.Digest) (name.Repository, error) {
 	var opts []name.Option
 	if cfg.Storage.OCI.Insecure {
 		opts = append(opts, name.Insecure)
 	}
 	if storageOCIRepository := cfg.Storage.OCI.Repository; storageOCIRepository != "" {
		return name.NewRepository(storageOCIRepository, opts...)
 	}
	return name.NewRepository(imageName.RepositoryStr(), opts...)
 }
```

That should do the job.